src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxKindsServiceFactory.cs

```diff
@@ -15,5 +14,10 @@ internal class CSharpSyntaxKindsServiceFactory : ILanguageServiceFactory
     {
         public ILanguageService CreateLanguageService(HostLanguageServices languageServices)
             => CSharpSyntaxKindsService.Instance;
+
+        private sealed class CSharpSyntaxKindsService : CSharpSyntaxKinds, ILanguageService
```

Comment:
`CSharpSyntaxKindsService` is now a mere implementation detail for language service based discovery of `ISyntaxKinds` instance in language agnostic layer. All clients in language specific layer can now use `CSharpSyntaxKinds.Instance`.